meumobi/sitebuilder
# Refactor/push flow

# Checklist

- [x] I successfully ran tests locally with my changes.
- [x] I've followed the guidelines from Contributing document.
- [x] I've updated wiki doc when necessary.
- [x] All expressions are translated .
- [x] All commits Close related issue, using `Closes #ISSUE_ID`.
- [x] Is Log Repository impacted ? if yes list changes and update Log Repository.
  - Changes on Log Repository should be related on Operating handbook and Scalyr.

## Tests
### Tests after refactor
1. Create item UI - push=true  // SHOULD CHECK/SEND PUSH
```
[2019-05-27 17:04:37] sitebuilder.INFO: "item created" {"item_id":"5cec18a510f0edb81f8b4567","site_id":3,"is_published":true,"category_id":15,"downloaded_images":0,"failed_images":0,"component":"create_item"}
[2019-05-27 17:04:37] sitebuilder.INFO: "job created" {"job_id":"5cec18a510f0edb81f8b4568","type":"process_remote_media","params":{"item_id":"5cec18a510f0edb81f8b4567"},"component":"worker_manager"}
[2019-05-27 17:04:37] sitebuilder.INFO: "job created" {"job_id":"5cec18a510f0edb81f8b4569","type":"push_notification","params":{"item_id":"5cec18a510f0edb81f8b4567","category_id":15},"component":"worker_manager"}
```
2. Create Item Feed - push=true - priority=high   // SHOULD NOT CHECK/NOT SEND PUSH
```
[2019-05-27 17:37:01] sitebuilder.DEBUG: item creation started {"guid":"https://api.spreaker.com/episode/17410511","category_id":16,"extension_id":"[object] (MongoId: 5cec1ff610f0edaf1f8b4568)","component":"update_news_feed"} []
[2019-05-27 17:37:01] sitebuilder.INFO: item created {"item_id":"5cec203d10f0ed45208b4579","site_id":3,"is_published":true,"category_id":16,"downloaded_images":0,"failed_images":0,"component":"create_item"} []
[2019-05-27 17:37:01] sitebuilder.INFO: job created {"job_id":"5cec203d10f0ed45208b457a","type":"process_remote_media","params":{"item_id":"5cec203d10f0ed45208b4579"},"component":"worker_manager"} []
[2019-05-27 17:37:01] sitebuilder.INFO: extension priority lowered {"extension_id":"[object] (MongoId: 5cec1ff610f0edaf1f8b4568)","category_id":16,"component":"update_news_feed"} []
```
3. Create Item Feed - push=true - priority=low   // SHOULD CHECK/SEND PUSH
```
[2019-05-27 17:36:22] sitebuilder.DEBUG: item creation started {"guid":"https://api.spreaker.com/episode/17410511","category_id":16,"extension_id":"[object] (MongoId: 5cec1ff610f0edaf1f8b4568)","component":"update_news_feed"} []
[2019-05-27 17:36:22] sitebuilder.INFO: item created {"item_id":"5cec201610f0ed3d208b4579","site_id":3,"is_published":true,"category_id":16,"downloaded_images":0,"failed_images":0,"component":"create_item"} []
[2019-05-27 17:36:22] sitebuilder.INFO: job created {"job_id":"5cec201610f0ed3d208b457a","type":"process_remote_media","params":{"item_id":"5cec201610f0ed3d208b4579"},"component":"worker_manager"} []
[2019-05-27 17:36:22] sitebuilder.DEBUG: notification is expired {"item_id":"5cec201610f0ed3d208b4579","publishedDate":"2019-03-23 00:48:55","limitDate":"2019-05-25 17:36:22","message":"published date is outdated","component":"push_notification"} []
[2019-05-27 17:36:22] sitebuilder.NOTICE: push not sent {"item_id":"5cec201610f0ed3d208b4579","message":"published date is outdated","component":"push_notification"} []
[2019-05-27 17:36:22] sitebuilder.INFO: feed checksum updated {"extension_id":"[object] (MongoId: 5cec1ff610f0edaf1f8b4568)","category_id":16,"component":"update_news_feed"} []
```

4. Create item UI - push=false  // SHOULD NOT CHECK/NOT SEND PUSH
```
[2019-05-27 17:56:18] sitebuilder.INFO: "item created" {"item_id":"5cec24c210f0edb31f8b4567","site_id":3,"is_published":true,"category_id":15,"downloaded_images":0,"failed_images":0,"component":"create_item"}
```

### Tests with Title
1. Create Item Title = !teste on UI , category-push=false// SHOULD CHECK/SEND PUSH
```
[2019-05-27 18:39:08] sitebuilder.INFO: "item created" {"item_id":"5cec2ecc10f0edb21f8b4569","site_id":3,"is_published":true,"category_id":15,"push":true,"downloaded_images":0,"failed_images":0,"component":"create_item"}
[2019-05-27 18:39:08] sitebuilder.INFO: "job created" {"job_id":"5cec2ecc10f0edb21f8b456b","type":"push_notification","params":{"item_id":"5cec2ecc10f0edb21f8b4569","category_id":15},"component":"worker_manager"}
```
2. Create Item Title = teste on UI , category-push=true// SHOULD CHECK/SEND PUSH
```
[2019-05-27 18:50:37] sitebuilder.INFO: "item created" {"item_id":"5cec317d10f0edaf1f8b4569","site_id":3,"is_published":true,"category_id":15,"downloaded_images":0,"failed_images":0,"component":"create_item"}
[2019-05-27 18:50:37] sitebuilder.INFO: "job created" {"job_id":"5cec317d10f0edaf1f8b456b","type":"push_notification","params":{"item_id":"5cec317d10f0edaf1f8b4569","category_id":15},"component":"worker_manager"}
```
3. Create Item Title = !teste on UI , category-push=true// SHOULD CHECK/SEND PUSH
```
[2019-05-27 18:50:47] sitebuilder.INFO: "item created" {"item_id":"5cec318710f0edb11f8b456c","site_id":3,"is_published":true,"category_id":15,"downloaded_images":0,"failed_images":0,"component":"create_item"}
[2019-05-27 18:50:47] sitebuilder.INFO: "job created" {"job_id":"5cec318710f0edb11f8b456e","type":"push_notification","params":{"item_id":"5cec318710f0edb11f8b456c","category_id":15},"component":"worker_manager"}
```

4.  Create Item Title = teste on UI , category-push=false// SHOULD NOT CHECK/NOT SEND PUSH
```
[2019-05-27 18:53:51] sitebuilder.INFO: "item created" {"item_id":"5cec323f10f0edae1f8b4567","site_id":3,"is_published":true,"category_id":15,"downloaded_images":0,"failed_images":0,"component":"create_item"}
```

5. Create Item Title = !teste on UI , category-push=false, published date on future
-  SHOULD NOT CHECK/NOT SEND PUSH WHEN CREATED
```
[2019-05-27 18:44:39] sitebuilder.INFO: "item created" {"item_id":"5cec301710f0edc11f8b4567","site_id":3,"is_published":false,"category_id":15,"downloaded_images":0,"failed_images":0,"component":"create_item"}
```
- SHOULD CHECK/SEND PUSH WHEN PUBLISHED
```
[2019-05-27 18:45:25] sitebuilder.INFO: item published {"item_id":"5cec2ff110f0edb91f8b4567","is_published":true,"component":"publish_items"} []
[2019-05-27 18:45:25] sitebuilder.INFO: job created {"job_id":"5cec304510f0edac208b4567","type":"push_notification","params":{"item_id":"5cec2ff110f0edb91f8b4567","category_id":15},"component":"worker_manager"} []
```
